Sources/BeaconChain/StateTransition.swift

```diff
@@ -280,6 +280,49 @@ extension StateTransition {
         }
     }
 
+    static func transfers(state: inout BeaconState, block: BeaconBlock) {
+        assert(block.body.transfers.count <= MAX_TRANSFERS)
+
+        for transfer in block.body.transfers {
+            assert(state.validatorBalances[Int(transfer.from)] >= transfer.amount)
+            assert(state.validatorBalances[Int(transfer.from)] >= transfer.fee)
+            assert(
+                state.validatorBalances[Int(transfer.from)] == transfer.amount + transfer.fee
+                || state.validatorBalances[Int(transfer.from)] >= transfer.amount + transfer.fee + MIN_DEPOSIT_AMOUNT
+            )
+
+            assert(state.slot == transfer.slot)
+            assert(BeaconChain.getCurrentEpoch(state: state) >= state.validatorRegistry[Int(transfer.from)].withdrawableEpoch)
+            // @todo does this work (-1)?
```

Comment:
Does it?

Reply:
If it does work, wouldn't it be `1 ..<= -1` so you include the last element?

Reply:
@djrtwo found something even nicer: https://developer.apple.com/documentation/foundation/data/1779685-suffix